core-common/src/main/java/org/zalando/nakadi/partitioning/PartitionStrategy.java

```diff
@@ -13,6 +22,53 @@ public interface PartitionStrategy {
     String USER_DEFINED_STRATEGY = "user_defined";
     String RANDOM_STRATEGY = "random";
 
-    String calculatePartition(EventType eventType, JSONObject event, List<String> partitions)
+    String calculatePartition(EventType eventType, PartitionData event, List<String> partitions)
             throws PartitioningException;
 }
+
+class PartitionData {
```

Comment:
I have implemented something roughly and more on a conceptual level. If you see build errors, please ignore them for now.